### PR TITLE
Fix cases where UTF8 wasn't supported.

### PR DESCRIPTION
**Pull request by yuehuang010**
- Fix an issue where .sln files are created without BOM fallback to asci.  This limits to just EN language. 
 Instead, fallback to UTF-8, which is a superset of asci.  The VS IDE by default creates sln in UFT8.
(UNDO)- When UseCommandProcessor is true, it will create a .cmd (aka .bat) file to run.  This file encoding is hardcoded for OS lang which is sometimes incorrect, but it doesn't provide a way to override.  Using StandardOutputEncoding as a override mechanism.

### CODE REVIEW

File: src/Build/Construction/Solution/SolutionFile.cs [hunk 1]
```diff
@@ -265,7 +265,7 @@ internal static void GetSolutionFileAndVisualStudioMajorVersions(string solution
             {
                 // Open the file
                 fileStream = File.OpenRead(solutionFile);
-                reader = new StreamReader(fileStream, Encoding.GetEncoding(0)); // HIGHCHAR: If solution files have no byte-order marks, then assume ANSI rather than ASCII.
+                reader = new StreamReader(fileStream, Encoding.UTF8); // HIGHCHAR: If solution files have no byte-order marks, then assume UFT8-8.
```
Comment (jgoshi) [inline src/Build/Construction/Solution/SolutionFile.cs]: nit UTF8 misspelled in the comment

Comment (yuehuang010) [reply src/Build/Construction/Solution/SolutionFile.cs]: Thanks. fixed.

File: src/Build/Construction/Solution/SolutionFile.cs [hunk 2]
```diff
@@ -265,7 +265,7 @@ internal static void GetSolutionFileAndVisualStudioMajorVersions(string solution
             {
                 // Open the file
                 fileStream = File.OpenRead(solutionFile);
-                reader = new StreamReader(fileStream, Encoding.GetEncoding(0)); // HIGHCHAR: If solution files have no byte-order marks, then assume ANSI rather than ASCII.
+                reader = new StreamReader(fileStream, Encoding.UTF8); // HIGHCHAR: If solution files have no byte-order marks, then assume UFT-8.
```
Comment (rainersigwald) [inline src/Build/Construction/Solution/SolutionFile.cs]:
```suggestion
                reader = new StreamReader(fileStream, Encoding.UTF8); // HIGHCHAR: If solution files have no byte-order marks, then assume UTF-8 (which also supports ASCII).
```

**Comment by rainersigwald** [comment]
Looks likehttps://github.com/microsoft/msbuild/blob/master/src/Build.UnitTests/Construction/SolutionFile_Tests.cs would be a good place for such a test.

**Comment by yuehuang010** [comment]
I have examined further regarding code page and how the VS IDE handles ambiguity.  A simple UFT8 isn't enough, sadly.  I am abandoning this PR.

**Pull request closed without merging**